### PR TITLE
Feature/first implementation

### PR DESCRIPTION
**Pull request by mchalapuk**

### CODE REVIEW

File: src/Aono.spec.coffee [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+  .default
+
+Aono = require "./Aono"
+  .default
+
+describe "Aono", ->
+  mocks =
+    timeProvider: sinon.stub()
+    handler0: sinon.stub()
+    handler1: sinon.stub()
```
Comment (mchalapuk) [inline src/Aono.spec.coffee:14]: `handler1` is not used

**Review by HubertBubert — APPROVED** [review]
Nice work !

**Review by mchalapuk — COMMENTED** [review]
@wgromniak ?

File: lib/Aono.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
+"use strict";
+var __extends = (this && this.__extends) || (function () {
```
Comment (wgromniak) [inline lib/Aono.js:2]: I started reviewing this file and was quite surprised about the coding style...

File: src/Logger.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ */
+export class Logger<Level> extends EventEmitter {
+  constructor(
+    private getTimestamp : TimeProvider,
+    private name : string,
+  ) {
+    super();
+  }
+
+  log(level : Level, message : string, meta : Object = {}) {
```
Comment (wgromniak) [inline src/Logger.ts:17]: What about convenience debug/info/warn/error? Are we planning to have some default string substitution mechanism for the `message`?

File: src/Logger.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { EventEmitter } from 'events';
+
+import Entry from './Entry';
+import TimeProvider from './TimeProvider';
+
+/**
+ * @author Maciej Chałapuk (maciej@chalapuk.pl)
+ */
+export class Logger<Level> extends EventEmitter {
```
Comment (wgromniak) [inline src/Logger.ts:9]: Why is it parametrised with `Level`?

**Review by wgromniak — APPROVED** [review]

**Comment by wgromniak** [comment]
Sorry for the tardy review...

File: src/Logger.ts (hunk 3 above)
Comment (mchalapuk) [reply src/Logger.ts:17]: Yes, I will add `DefaultLevel` and `DefaultLogger` later. As for string substitution, I think it's not needed when we have ES6 template strings.

File: src/Logger.ts (hunk 4 above)
Comment (mchalapuk) [reply src/Logger.ts:9]: I didn't want to hardcode particular levels at this level of abstraction.

**Comment by wgromniak** [comment]
ACK to your replies.

sent from mobile

**Pull request merged**